tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/service/TaskService.java

```diff
@@ -292,7 +292,12 @@ private void updateCompletedMetric(final TaskEntity task) {
     try {
       metrics.recordCounts(COUNTER_NAME_COMPLETED_TASKS, 1, getTaskMetricLabels(task));
       assigneeMigrator.migrateUsageMetrics(getCurrentUser().getUserId());
-      taskMetricsStore.registerTaskCompleteEvent(task);
+      // Only write metrics when completing a Job-based User Tasks. With 8.7,
+      // metrics for completed (not Job-based) User Tasks are written by the
+      // handler "TaskCompletedMetricHandler" in the camunda-exporter
+      if (task.getImplementation().equals(TaskImplementation.JOB_WORKER)) {
```

Comment:
:thinking: @romansmirnov Would it make sense to move this to the Exporter as well to be consistent? We can capture the completion intent for Job based User Tasks.

Reply:
No, we can't because the job in the engine might not have the last assigned user set. The assignment of Job-based User Tasks happens in Tasklist itself, that's why they must also write the metric for those user tasks.